frontend/src/pages/shipments/ShipmentList.tsx

```diff
@@ -0,0 +1,135 @@
+import { gql, useQuery } from '@apollo/client'
+import cx from 'classnames'
+import React, { FunctionComponent, useMemo } from 'react'
+import { useSortBy, useTable } from 'react-table'
+import Badge from '../../components/Badge'
+import TableHeader from '../../components/table/TableHeader'
+import LayoutWithNav from '../../layouts/LayoutWithNav'
+import { Shipment } from '../../types/api-types'
+import {
+  formatLabelMonth,
+  getShipmentStatusBadgeColor,
+} from '../../utils/format'
+
+const SHIPMENTS_QUERY = gql`
+  query GetAllShipments {
+    listShipments {
+      id
+      shippingRoute
+      labelYear
+      labelMonth
+      offerSubmissionDeadline
+      status
+      sendingHub {
+        id
+        name
+      }
+      receivingHub {
+        id
+        name
+      }
+      statusChangeTime
+    }
+  }
+`
+
+const COLUMNS = [
+  {
+    Header: 'ID',
+    accessor: 'id',
+  },
+  {
+    Header: 'Route',
+    accessor: 'shippingRoute',
+  },
+  {
+    Header: 'Sending hub',
+    accessor: 'sendingHub.name',
+  },
+  {
+    Header: 'Receiving hub',
+    accessor: 'receivingHub.name',
+  },
+  {
+    Header: 'Date',
+    accessor: (row: Shipment) =>
+      `${formatLabelMonth(row.labelMonth)} ${row.labelYear}`,
+  },
+  {
+    Header: 'Status',
+    accessor: 'status',
+    Cell: ({ value }: any) => (
+      <Badge color={getShipmentStatusBadgeColor(value)}>{value}</Badge>
+    ),
+  },
+]
+
+const ShipmentList: FunctionComponent = () => {
+  const { data } = useQuery<{ listShipments: Shipment[] }>(SHIPMENTS_QUERY)
+
+  // We must memoize the data for react-table to function properly
```

Comment:
Question (non-blocking): if we don't do this, what happens?

Reply:
BAD THINGS, BION. It re-rendered infinitely until the server crashed haha

Reply:
Haaah well that's obvious enough. Mostly curious how easy it would be for someone to inadvertently introduce a bug by not memoizing.

Reply:
The docs repeat "please memoize your data" everywhere, which helps.